sys/include/net/skald.h

```diff
@@ -65,70 +66,20 @@ extern "C" {
 #ifndef CONFIG_SKALD_INTERVAL
 #define CONFIG_SKALD_INTERVAL          (1 * US_PER_SEC)
 #endif
-
-/**
- * @brief   Configure advertising channel 37
- *
- * Set CONFIG_ADV_CH_37_DISABLE to disable channel 37
- */
-#ifdef DOXYGEN
-#define CONFIG_ADV_CH_37_DISABLE
-#endif
-
-/**
- * @brief   Configure advertising channel 38
- *
- * Set CONFIG_ADV_CH_38_DISABLE to disable channel 38
- */
-#ifdef DOXYGEN
-#define CONFIG_ADV_CH_38_DISABLE
-#endif
-
-/**
- * @brief   Configure advertising channel 39
- *
- * Set CONFIG_ADV_CH_39_DISABLE to disable channel 39
- */
-#ifdef DOXYGEN
-#define CONFIG_ADV_CH_39_DISABLE
-#endif
 /** @} */
 
-/**
- * @brief   Define advertising channel 37 if @ref CONFIG_ADV_CH_37_DISABLE is
- *          not set
- */
-#if !defined(CONFIG_ADV_CH_37_DISABLE) || defined(DOXYGEN)
-#define ADV_CH_37 37,
-#else
-#define ADV_CH_37
-#endif
-
-/**
- * @brief   Define advertising channel 38 if @ref CONFIG_ADV_CH_38_DISABLE is
- *          not set
- */
-#if !defined(CONFIG_ADV_CH_38_DISABLE) || defined(DOXYGEN)
-#define ADV_CH_38 38,
-#else
-#define ADV_CH_38
-#endif
-
-/**
- * @brief   Define advertising channel 39 if @ref CONFIG_ADV_CH_39_DISABLE is
- *          not set
- */
-#if !defined(CONFIG_ADV_CH_39_DISABLE) || defined(DOXYGEN)
-#define ADV_CH_39 39
-#else
-#define ADV_CH_39
-#endif
-
 /**
  * @brief   List of advertising channels
  */
 #ifndef SKALD_ADV_CHAN
-#define SKALD_ADV_CHAN { ADV_CH_37 ADV_CH_38 ADV_CH_39 }
+#define SKALD_ADV_CHAN { ADV_CH_37 ADV_CH_0 ADV_CH_1 ADV_CH_2 ADV_CH_3 \
+                         ADV_CH_4 ADV_CH_5 ADV_CH_6 ADV_CH_7 ADV_CH_8 ADV_CH_9 \
+                         ADV_CH_10 ADV_CH_38 ADV_CH_11 ADV_CH_12 ADV_CH_13 \
+                         ADV_CH_15 ADV_CH_16 ADV_CH_17 ADV_CH_18 ADV_CH_19 \
+                         ADV_CH_20 ADV_CH_21 ADV_CH_22 ADV_CH_23 ADV_CH_24 \
+                         ADV_CH_25 ADV_CH_26 ADV_CH_27 ADV_CH_28 ADV_CH_29 \
+                         ADV_CH_30 ADV_CH_31 ADV_CH_32 ADV_CH_33 ADV_CH_34 \
+                         ADV_CH_35 ADV_CH_36 ADV_CH_38 ADV_CH_39 }
```

Comment:
What about instead of this, you have another definition of the array in `skald_kconfig.h` which only applies when Kconfig is used. You can just leave this default as before with the basic channels. In the form of:
```c
{
#if IS_ACTIVE(CONFIG_ADV_CH_1_ENABLE)
1,
#endif
#if IS_ACTIVE(CONFIG_ADV_CH_2_ENABLE)
2,
#endif
[...]
}
```
Instead of all those extra macros?

Reply:
Yes, makes sense :) . Thanks for the suggestion.

Reply:
Done. Multiline `#defines` with conditions seems to behave strangely hence kept the syntax intact.

Reply:
What do you mean?